src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,77 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
+@Component
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
```

Comment:
```suggestion
    private Map<Class<?>, Object> instances = new HashMap<>();
```
Maybe it's better to initialize this in the constructor (method getInstance())?

Reply:
Не знаю, вот на счет мапы с метода findImplementation может ее лучше вынести как поле класса, потому что она получается постоянно создается при вызове соответствующего метода. Но ее можно создать один раз.